File: 01.hello/Hello2.java
public class Hello2 {
  public static void main(String[] args) {
    if (args.length == 2) {
      String greeting = String.format("Hello %s %s!", args[0], args[1]);
      System.out.println(greeting);
    }
    else
      System.out.println("Hello World!");
  }
}

File: 01.hello/hello2.ml
let main (args : string array) =
  if (Array.length args = 3) then
    let greeting : string = Printf.sprintf "Hello %s %s!" args.(1) args.(2)
    in
    print_endline greeting
  else
    print_endline "Hello World!"

let _ = main Sys.argv























(*
let main args =
  if Array.length args = 3 then
    let greeting = Printf.sprintf "Hello %s %s!" args.(1) args.(2)
    in
    print_endline greeting
  else
    print_endline "Hello World!"

let _ = main Sys.argv
*)
